### PR TITLE
fix(ipnet): add ip to unsmarshall and test

### PR DESCRIPTION
**Pull request by Monitob**
- fix(custom_types): ipnet unmarshal json
- Revert "fix(custom_types): ipnet unmarshal json"
- fix(custom_types): ipnet unmarshal json

### CODE REVIEW

File: scw/custom_types_test.go [hunk 1]
```diff
@@ -343,6 +343,11 @@ func TestIPNet_MarshallJSON(t *testing.T) {
 			ipRange: IPNet{IPNet: net.IPNet{IP: net.IPv4(42, 42, 42, 42), Mask: net.CIDRMask(16, 32)}},
 			want:    `"42.42.42.42/16"`,
 		},
+		{
+			name:    "network ip2",
```
Comment (kindermoumoute) [inline scw/custom_types_test.go:347]:
```suggestion
			name:    "network with ip",
```

File: api/instance/v1/security_group_utils_test.go [hunk 2]
```diff
@@ -137,7 +137,7 @@ func TestAPI_UpdateSecurityGroupRule(t *testing.T) {
 
 		testhelpers.AssertNoError(t, err)
 		testhelpers.Equals(t, SecurityGroupRuleActionDrop, updateResponse.Rule.Action)
-		testhelpers.Equals(t, scw.IPNet{IPNet: net.IPNet{IP: net.IP{0x1, 0x1, 0x1, 0x1}, Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, updateResponse.Rule.IPRange)
+		testhelpers.Equals(t, scw.IPNet{IPNet: net.IPNet{IP: net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0x1, 0x1, 0x1, 0x1}, Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, updateResponse.Rule.IPRange)
```
Comment (kindermoumoute) [inline api/instance/v1/security_group_utils_test.go:140]: Why do you need an IPv6 here?

File: scw/custom_types.go [hunk 3]
```diff
@@ -238,11 +238,12 @@ func (n *IPNet) UnmarshalJSON(b []byte) error {
 		str += "/128"
 	}
 
-	_, value, err := net.ParseCIDR(str)
+	ip, value, err := net.ParseCIDR(str)
 	if err != nil {
 		return err
 	}
 	n.IPNet = *value
+	n.IP = ip
```
Comment (kindermoumoute) [inline scw/custom_types.go:246]: does something like that work too?
```suggestion
	value.IP = ip
	n.IPNet = *value
```

**Review by kindermoumoute — CHANGES_REQUESTED** [review]

File: api/instance/v1/security_group_utils_test.go (hunk 2 above)
Comment (Monitob) [reply api/instance/v1/security_group_utils_test.go:140]: I just add the missing bytes that are returned now

Comment (kindermoumoute) [reply api/instance/v1/security_group_utils_test.go:140]: These bytes are corresponding to an IPv6, not an IPv4

File: api/instance/v1/security_group_utils_test.go [hunk 4]
```diff
@@ -210,7 +210,7 @@ func TestAPI_UpdateSecurityGroupRule(t *testing.T) {
 
 		testhelpers.AssertNoError(t, err)
 		testhelpers.Equals(t, SecurityGroupRuleActionAccept, updateResponse.Rule.Action)
-		testhelpers.Equals(t, scw.IPNet{IPNet: net.IPNet{IP: net.IP{0x8, 0x8, 0x8, 0x8}, Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, updateResponse.Rule.IPRange)
+		testhelpers.Equals(t, scw.IPNet{IPNet: net.IPNet{IP: net.IP{0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0xff, 0xff, 0x8, 0x8, 0x8, 0x8}, Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, updateResponse.Rule.IPRange)
```
Comment (kindermoumoute) [inline api/instance/v1/security_group_utils_test.go:213]: Use this notation

```suggestion
		testhelpers.Equals(t, scw.IPNet{IPNet: net.IPv4(0x8, 0x8, 0x8, 0x8), Mask: net.IPMask{0xff, 0xff, 0xff, 0xff}}}, updateResponse.Rule.IPRange)
```

**Review by kindermoumoute — APPROVED** [review]
LGTM

File: api/instance/v1/security_group_utils_test.go (hunk 2 above)
Comment (Monitob) [reply api/instance/v1/security_group_utils_test.go]: I've juste apply net.IPv4 to be more explanatory.

**Pull request merged**